### PR TITLE
Patch 10

### PR DESCRIPTION
**Pull request by SnapDragon7410**

### CODE REVIEW

File: userbot/__main__.py [hunk 1]
```diff
@@ -25,6 +25,7 @@
 
 for module_name in ALL_MODULES:
     imported_module = import_module("userbot.modules." + module_name)
+import_module("userbot.plug_core")
```
Comment (penn5) [inline userbot/__main__.py]: Please move this to ALL_MODULES

File: userbot/events.py [hunk 2]
```diff
@@ -12,14 +12,20 @@
 from os import remove
 from time import gmtime, strftime
 from traceback import format_exc
+from pathlib import Path
 
 from telethon import events
+import stack
```
Comment (penn5) [inline userbot/events.py]: inspect not stack

File: userbot/events.py [hunk 3]
```diff
@@ -51,6 +57,11 @@ def register(**args):
             args['pattern'] = pattern.replace('^.', unsafe_pattern, 1)
 
     def decorator(func):
+        try:
+            LOAD_PLUG[file_test].append(func)
+        except:
+            LOAD_PLUG.update({file_test: []})
+            LOAD_PLUG[file_test].append(func)
```
Comment (penn5) [inline userbot/events.py]: Combine these two lines into one:

`LOAD_PLUG[file_test] = [func]`

File: userbot/plug_core.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+from userbot.events import register
+
+@register(pattern="^.unload (?P<shortname>\w+)$", outgoing=True)
+async def unload(event):
+    if event.fwd_from:
+        return
+    shortname = event.pattern_match["shortname"]
+    try:
+        remove_plugin(shortname)
+        await event.edit(f"Unloaded {shortname} successfully")
```
Comment (penn5) [inline userbot/plug_core.py]: Put this in an `else` clause

File: userbot/__main__.py (hunk 1 above)
Comment (SnapDragon7410) [reply userbot/__main__.py]: must i `append` it?

File: userbot/events.py (hunk 2 above)
Comment (SnapDragon7410) [reply userbot/events.py]: ok

File: userbot/plug_core.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    if event.fwd_from:
+        return
+    shortname = event.pattern_match["shortname"]
+    try:
+        try:
+            remove_plugin(shortname)
+        except:
+            pass
+        import_module(f"userbot.modules.{shortname}")
+        await event.edit(f"Successfully loaded {shortname}")
```
[the remark this answers is not in the document]
Comment (SnapDragon7410) [reply userbot/plug_core.py]: example?

File: userbot/plug_core.py (hunk 4 above)
Comment (SnapDragon7410) [reply userbot/plug_core.py]: example?

File: userbot/plug_core.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+async def load(event):
+    if event.fwd_from:
+        return
+    shortname = event.pattern_match["shortname"]
+    try:
+        try:
+            remove_plugin(shortname)
+        except:
+            pass
+        import_module(f"userbot.modules.{shortname}")
```
[the remark this answers is not in the document]
Comment (SnapDragon7410) [reply userbot/plug_core.py]: hmm

File: userbot/modules/plug_core.py [hunk 7]
```diff
@@ -15,8 +15,9 @@
     try:
         remove_plugin(shortname)
         await event.edit(f"Unloaded {shortname} successfully")
-    except Exception as e:
-        await event.edit("Could not unload {} due to the following error.\n{}".format(shortname, str(e)))
+    except KeyError:
+        await event.edit(f"{shortname} does not exist in your modules")
+        raise KeyError("{} is not a module".format(shortname))
```
Comment (penn5) [inline userbot/modules/plug_core.py]: wth? this will trigger a bug report

Comment (penn5) [reply userbot/modules/plug_core.py]: and also its not that it doesnt exist, but that its not loaded

File: userbot/modules/plug_core.py [hunk 8]
```diff
@@ -30,8 +31,9 @@
             pass
         import_module(f"userbot.modules.{shortname}")
         await event.edit(f"Successfully loaded {shortname}")
-    except Exception as e:
-        await event.edit(f"Could not load {shortname} because of the following error.\n{str(e)}")
+    except KeyError:
+        await event.edit(f"{shortname} does not exist in your modules")
+        raise KeyError("{} is not a module".format(shortname))
```
Comment (penn5) [inline userbot/modules/plug_core.py]: again wth

**Pull request closed without merging**